haystack/nodes/reader/table.py

```diff
@@ -320,18 +315,33 @@ def _calculate_answer_score(
             "inv_column_ranks",
             "numeric_relations",
         ]
+        segment_ids = inputs["token_type_ids"][:, :, token_types.index("segment_ids")]
```

Comment:
Same code as before in `_calculate_answer_score` except set up to work with tensors that have a batch dimension. Even for `batch_size=1` the batch dimension is kept, so the logic of this code is checked in the tests since `batch_size=1` is default.